components/org.wso2.carbon.identity.password.history/src/main/java/org/wso2/carbon/identity/password/history/listener/PasswordHistoryTenantMgtListener.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.password.history.listener;
+
+import org.wso2.carbon.identity.password.history.exeption.IdentityPasswordHistoryException;
+import org.wso2.carbon.identity.password.history.store.Impl.DefaultPasswordHistoryDataStore;
+import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
+import org.wso2.carbon.stratos.common.exception.StratosException;
+import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
+
+public class PasswordHistoryTenantMgtListener implements TenantMgtListener {
```

Comment:
Isn't there an abstract listener?

Reply:
org.wso2.carbon.identity.core.AbstractIdentityTenantMgtListener

Reply:
Use this one instead of implementing from TenantMgtListener